### PR TITLE
Implementation of dissipation balanced convective penetration

### PR DESCRIPTION
**Pull request by mathren**
From [Johnston et al. 2024 ](https://ui.adsabs.harvard.edu/abs/2024ApJ...964..170J/abstract) informed by the 3D simulation by [Anders et al. 2022](https://ui.adsabs.harvard.edu/abs/2022ApJ...926..169A/abstract).

I tested that my re-implementation gives qualitative and quantitative agreement with the original calculations of an 8Msun, solar Z, main sequence star. The plot below shows the mass in the convective penetration zone as a function of time, with in blue the original implementation and inlist (labeled "baseline") , and in orange the output from the example folder in `dbconvpen` (labeled "example rei-mplementation". The two curves are not expected to be numerically identical since I did try to remove some things from the example inlist to simplify, but they are expected to be physically equivalent, and indeed appear:

image

 Implementing this in `mesa-contrib` was pre-approved by @colej. Worth tagging also @evananders, @MichielsenM in case they want to chime in.

### CODE REVIEW

**Comment by warrickball** [comment]
This looks good but I'm afraid I probably won't be able to test it until at least the end of July. I hope that's not too long to wait!

The only thing I did notice was a couple of old-fashioned comparison operators (e.g. `.lt.` and `.le.`) even though modern style comparisons (e.g. `>=`) are used elsewhere in the code.

**Comment by mathren** [comment]
That is fixed now! And no worries, there is no rush at all

File: hooks/dbconvpen/dbconvpen.inc [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+
+    s% lxtra(1) = .true.
+    V_CZ = 0d0
+    Lint = 0d0
+
+    call star_eval_conv_bdy_k(s, 1, k, ierr)
+    call star_eval_conv_bdy_r(s, 1, r_cb, ierr)
+    s% ixtra(2) = k
+    s% xtra(5) = r_cb
+    s%xtra(1) = s%m(k)
```
Comment (warrickball) [inline hooks/dbconvpen/dbconvpen.inc]:
```suggestion
    s% xtra(1) = s%m(k)
```

**Comment by warrickball** [comment]
There's a bit of re-arranging I can do once this is merged but I'm satisfied with the overall quality. Some cleanup I'll do later is

* rename `example` to `star_example` (so that it'll work with `check_examples.sh`),
* remove files in `example` that are already part of the work folder, and
* maybe try to reduce the example run time (took me  on 4 cores).

Out of curiosity, do you now have permission to merge this yourself?

**Pull request merged**